### PR TITLE
Automatically convert type arguments into RunInput subclasses

### PR DESCRIPTION
**Pull request by abrookins**
This PR makes writing interactive flows even easier:
 
1. If you pause or suspend a flow run to receive a single-value input (e.g., `age: int`, `name: str`, etc.), Prefect will create the `RunInput` class for you.
2. Similarly, you can now send single-value inputs to flow runs and receive them in the same manner without creating a `RunInput` class. 
3. You can now use your existing `pydantic.BaseModel` classes without any modification. Any function that takes a `RunInput` can instead take a `pydantic.BaseModel`, and Prefect will create a `RunInput` from your `BaseModel`.

## Automatic inputs for paused and suspended flows
Pausing and suspending to wait for single-value run inputs does not require creating a `RunInput` class. Just pass the type of the value you want to receive:

```python
age = await pause_flow_run(int, timeout=10, poll_interval=2)
```

Previously, you would have had to write something like the following:

```python
class AgeInput(RunInput):
    age: int

input = await pause_flow_run(AgeInput, timeout=10, poll_interval=2)
input.age
```

**NOTE:** You can still use an explicit `RunInput` class to change things like the name of this field when it appears in our UI, or collect more than one value.

## Automatic inputs for interactive flows
You can also use automatic input conversion with our sending and receiving functions (new: `send_input` and `receive_input`), without using pause/suspend:

```python
# Flow one
await send_input(10, flow_run_id=123)

# Flow two
age = await receive_input(int)
```

## Use your existing `BaseModel` classes as input
Finally, if you have an existing `BaseModel` and don't need to change any behavior, you can now pass it to any function that accepts a `RunInput` and we will do the needful to make it work for flow run input.

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] This pull request includes tests or only affects documentation.
- [ ] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [ ] This pull request includes helpful docstrings.
- [ ] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `mkdocs.yml` navigation.

### CODE REVIEW

**Comment by abrookins** [comment]
I'm not 100% I've done the typing here correctly: https://github.com/PrefectHQ/prefect/pull/11636/files#diff-ae6f1b6d423c9f28188aedf4ab0f0afbce428d61b84feed78ca9c716ce3e26caR978

**Comment by bunchesofdonald** [comment]
> I'm not 100% I've done the typing here correctly: https://github.com/PrefectHQ/prefect/pull/11636/files#diff-ae6f1b6d423c9f28188aedf4ab0f0afbce428d61b84feed78ca9c716ce3e26caR978

I think you just want to change the `TypeVar` `T` to not be bounded. So you'd just want to update them from:

`T = TypeVar("T", bound=RunInput)` to `T = TypeVar("T")` I think we define that both in the engine and in `run_input.py`.

File: src/prefect/input/run_input.py
Comment (bunchesofdonald) [inline src/prefect/input/run_input.py:299]: You could assign this a typevar (like we use `T`) and then you could load's return type be that typevar, I _think_ that would give nice type handling, so if a user were to pass `int` through it should come back out as `int` in their code. You might also need to update `AutomaticRunInput` to be `AutomaticRunInput[V]` in `subclass_from_type` and make this class a `Generic[V]`. If that all sounds messy I can also just play with it after this is merged and get it working.

**Review by bunchesofdonald — APPROVED** [review]
Looks great! I think the typing needs some help, but I also don't think that should block this from being merged.

**Pull request merged**
